api/src/main/java/io/opentelemetry/OpenTelemetry.java

```diff
@@ -86,68 +100,86 @@ public static MeterFactory getMeterFactory() {
+   * the global singleton will be initialized with an SPI-provided implementation.
+   *
+   * <p>Therefore, if you wish to use a non-SPI provided instance, you must make sure to call this
+   * method before any instrumentation will have forced the SPI implementation to be loaded.
+   *
+   * @param meterFactory A fully configured and ready to operate MeterFactory implementation.
+   */
+  public static void setMeterFactory(MeterFactory meterFactory) {
+    if (!instance.meterFactory.compareAndSet(null, meterFactory)) {
+      logger.warning(
+          "The global OpenTelemetry MeterFactory instance has already been set. "
+              + "Ignoring this assignment");
     }
-    return instance;
   }
 
-  private OpenTelemetry() {
-    TracerFactoryProvider tracerFactoryProvider = loadSpi(TracerFactoryProvider.class);
-    this.tracerFactory =
-        tracerFactoryProvider != null
-            ? tracerFactoryProvider.create()
-            : DefaultTracerFactoryProvider.getInstance().create();
+  /**
+   * Assigns the global singleton TracerFactory instance. This can be done exactly once; subsequent
+   * calls will be ignored.
+   *
+   * <p>Note, if calls to the static accessor for the TracerFactory are made before this is called,
+   * the global singleton will be initialized with an SPI-provided implementation.
+   *
+   * <p>Therefore, if you wish to use a non-SPI provided instance, you must make sure to call this
+   * method before any instrumentation will have forced the SPI implementation to be loaded.
+   *
+   * @param tracerFactory A fully configured and ready to operate {@link TracerFactory}
+   *     implementation.
+   */
+  public static void setTracerFactory(TracerFactory tracerFactory) {
+    if (!instance.tracerFactory.compareAndSet(null, tracerFactory)) {
+      logger.warning(
+          "The global OpenTelemetry TracerFactory instance has already been set. "
+              + "Ignoring this assignment");
+      return;
+    }
 
-    MeterFactoryProvider meterFactoryProvider = loadSpi(MeterFactoryProvider.class);
-    meterFactory =
-        meterFactoryProvider != null
-            ? meterFactoryProvider.create()
-            : DefaultMeterFactoryProvider.getInstance().create();
-    DistributedContextManagerProvider contextManagerProvider =
-        loadSpi(DistributedContextManagerProvider.class);
-    contextManager =
-        contextManagerProvider != null
-            ? contextManagerProvider.create()
-            : DefaultDistributedContextManager.getInstance();
+    // swap out any tracers that have been handed out with the proper ones.
+    Map<TracerKey, DefaultTracer> existingTracers = DefaultTracerFactory.getExistingTracers();
+    for (Entry<TracerKey, DefaultTracer> entry : existingTracers.entrySet()) {
+      TracerKey key = entry.getKey();
+      entry.getValue().setImplementation(tracerFactory.get(key.getName(), key.getVersion()));
+    }
```

Comment:
@bogdandrutu here is where the tracer implementation is swapped out after the SDK has been initialized.